### ISSUE TITLE
fix buflist.format.buffer and buflist.format.buffer_current for a buflist item in a window bar

### ISSUE THREAD

**Issue by pascalpoitras** [weechat/weechat#1542]
## Bug summary

Suppose you have two windows. The first window is the active window and is currently displaying the buffer, core.weechat, for example. The second window is displaying the buffer irc.freenode.#weechat and this window is displaying a window bar and this window bar is displaying the buflist3 item. Even if irc.freenode.#weechat is the buffer displayed in the second window, when the buflist3 item is processed and this is the turn of irc.freenode.#weechat to be processed, this is the option buflist.format.buffer that is used not buflist.format.buffer_current (only the buffer core.weechat will be processed by buffer_current because this is the active buffer in the active window (window 1)). The workaround is to check if the buffer is the same as window.buffer in both options. For example

```
/set buflist.format.buffer "${if:${bar_item.name}==buflist3?${if:${window.buffer}==${buffer}?${color:31}:${color:white}}${name}}"
/set buflist.format.buffer_current "${format_buffer}"
```

it would be better if buflist.format.buffer_current was used for the active buffer in window 2 and buflist.format.buffer for all other buffers

this is not critical, the workaround works fine but I would prefer to use something like this

```
/set buflist.format.buffer "${if:${bar_item.name}==buflist3?${if:${current_buffer}?${color:31}:${color:white}}${name}}"
/set buflist.format.buffer_current "${format_buffer}"
```

TL;DR

for a buflist item in a window bar, use buflist.format.buffer_current for the active buffer in this window not the active buffer of the active window. use buflist.format.buffer for all buffers not active in this window

---

- WeeChat version: 2.9
- OS, distribution and version: Debian 10
